packages/machine-extractor/src/extractAction.ts

```diff
@@ -351,23 +268,16 @@ export const getActionCreatorName = (actionNode: ActionNode) =>
     ? actionNode.node.callee.name
     : undefined;
 
+// TODO: enforce return type to ActionEvent so `type` property is required
 function extractEventObject(
   eventObject: t.ObjectExpression,
   fileContent: string,
-):
-  | { type: 'expression'; value: string }
-  | {
-      type: 'object';
-      value: Record<string, JsonEntry>;
-    } {
+): JsonExpressionString | Record<string, JsonItem> {
   if (!eventObject.properties.every((prop) => t.isObjectProperty(prop))) {
-    return {
-      type: 'expression',
-      value: fileContent.slice(eventObject.start!, eventObject.end!),
-    };
+    return `{{${fileContent.slice(eventObject.start!, eventObject.end!)}}}`;
```

Comment:
I'd introduce some util, like `asExpressionString` or smth. It's hard to tell here if we have the right amount of curly braces at the end since there are 3 (it's correct here, but it's an easy mistake to make)